theseus/optimizer/autograd/baspacho_sparse_autograd.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+
+from ..linear_system import SparseStructure
+
+from scipy.sparse import csr_matrix, csc_matrix
+import numpy as np
+
+
+def mat_vec_cpu(batch_size, num_cols, A_rowPtr, A_colInd, A_val, v):
```

Comment:
I just realized that many files like these in `theseus/optimizer/` do not have type hints (input types, return type) included when writing functions.

Reply:
+1.

Reply:
Good catch. Let's make an issue for this. Definitely not to be solved in this PR.